Fix translation: gatsby-on-linux.md (#238)
gatsbyjs/gatsby-pt-BR: docs/docs/gatsby-on-linux.md

title: Gatsby no Linux
---

> Este guia assume que você já possui uma instalação nativa do Linux em sua máquina. As etapas a seguir mostram como instalar o Node.js e dependências associadas.

Reviewer: ```suggestion
Este guia assume que você já possui uma instalação nativa do Linux em sua máquina. As etapas a seguir mostram como instalar o Node.js e suas dependências associadas.
```

> ## Ubuntu, Debian, e outras distros baseadas no `apt`

Reviewer 2: ```suggestion
## Ubuntu, Debian, e outras distribuições baseadas no `apt`
```

> Comece dando update e upgrade nos pacotes já existentes.

Reviewer 2: ```suggestion
Comece dando `update` e `upgrade` nos pacotes já existentes.
```

Author: Aqui eu fiquei em dúvida se colocava apenas "atualize seus pacotes" ou se deixava do jeito que deixei. O que tu acha @henry-ns ?

Reviewer 2: assim, pq está se referindo aos comandos

> Comece dando `update` e `upgrade` nos pacotes já existentes.

Reviewer: ```suggestion
Comece rodando `update` e `upgrade` nos pacotes já existentes.
```

```shell
sudo apt update
sudo apt -y upgrade
```

Instale o cURL que permite transferência de dados e o download de dependências adicionais.

```shell
sudo apt install curl
```

Agora você pode usar o `curl` para instalar o `nvm`, que irá gerenciar a instalação do `node` e todas suas versões associadas.

```shell
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.1/install.sh | bash
```

> Observe que esta é a versão estável atual do nvm. A instalação completa e resolução de problemas podem ser encontradas na [página do nvm no GitHub](https://github.com/nvm-sh/nvm)

> O `nvm` não instala uma versão do `node`. Então você precisa instalar a versão uma versão e fornecer ao `nvm` as instruções para utilizá-lo. Este exemplo usa o último lançamento da versão `10`, mas novas versões podem ser usadas também.

Reviewer: ```suggestion
O `nvm` não instala uma versão do `node`. Então você precisa instalar a versão que você quer e fornecer ao `nvm` as instruções para utilizá-la como padrão. Este exemplo usa o último lançamento da versão `10`, mas novas versões podem ser usadas também.
```

```shell
nvm install 10
nvm use 10
```

Para ter certeza que a instalação foi bem sucedida, use o seguinte comando.

```shell
node -v
```

> Observe que `npm` vem no mesmo pacote que o `node`

Por último, instale o `git` que vai ser necessário para criar seu primeiro projeto do Gatsby baseado em template `starter`

```shell
sudo apt install git
```

> ## Fedora, RedHat e outras distros baseadas no `dnf`

Reviewer 2: ```suggestion
## Fedora, RedHat e outras distribuições baseadas no `dnf`
```

> Estas distros já vem com o `curl` instalado, já podemos usá-lo para baixar o `nvm`

Reviewer: ```suggestion
Estas distros já vem com o `curl` instalado, então já podemos usá-lo para baixar o `nvm`
```

```shell
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.1/install.sh | bash
```

> Observe que esta é a versão estável atual do nvm. A instalação completa e resolução de problemas podem ser encontradas na [página do nvm no GitHub](https://github.com/nvm-sh/nvm)

O `nvm` não instala uma versão do `node`. Então você precisa instalar a versão uma versão e fornecer ao `nvm` as instruções para utilizá-lo. Este exemplo usa o último lançamento da versão `10`, mas novas versões podem ser usadas também.

```shell
nvm install 10
nvm use 10
```

Para ter certeza que a instalação foi bem sucedida, use o seguinte comando.

```shell
node -v
```

> Observe que `npm` vem no mesmo pacote que o `node`

Por último, instale o `git` que vai ser necessário para criar seu primeiro projeto do Gatsby baseado em template `starter`

```shell
sudo dnf install git
```

> ## Archlinux e outras distros baseadas em `pacman`

Reviewer 2: ```suggestion
## Archlinux e outras distribuições baseadas em `pacman`
```

> Comece dando atualizando os pacotes já existentes

Reviewer: ```suggestion
Comece atualizando os pacotes já existentes
```

```shell
sudo pacman -Sy
```

> Estas distros já vem com o `curl` instalado, já podemos usá-lo para baixar o `nvm`

Reviewer: ```suggestion
Estas distros já vem com o `curl` instalado, então já podemos usá-lo para baixar o `nvm`.
```

```shell
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.1/install.sh | bash
```

> Observe que esta é a versão estável atual do nvm. A instalação completa e resolução de problemas podem ser encontradas na [página do nvm no GitHub](https://github.com/nvm-sh/nvm)

Antes de usar o `nvm`, você precisa instalar as dependências adicionais.

```shell
sudo pacman -S grep awk tar git
```

O `nvm` não instala uma versão do `node`. Então você precisa instalar a versão uma versão e fornecer ao `nvm` as instruções para utilizá-lo. Este exemplo usa o último lançamento da versão `10`, mas novas versões podem ser usadas também.

```shell
nvm install 10
nvm use 10
```

Para ter certeza que a instalação foi bem sucedida, use o seguinte comando.

```shell
node -v
```

> Observe que `npm` vem no mesmo pacote que o `node`

## Windows Subsystem Linux (WSL)

> Este guia pressupõe que você já possui o WSL instalado com uma distribuição Linux em funcionamento. Caso não possua, siga as instruções [deste guia da Microsoft](https://docs.microsoft.com/en-us/windows/wsl/install-win10) para instalar o WSL e uma distro linux da sua escolha

Reviewer: ```suggestion
Este guia pressupõe que você já possui o WSL instalado com uma distribuição Linux em funcionamento. Caso não possua, siga as instruções [deste guia da Microsoft](https://docs.microsoft.com/pt-br/windows/wsl/install-win10) para instalar o WSL e uma distro linux de sua escolha.
```

sudo apt update && sudo apt -y upgrade && sudo apt install build-essential && sudo apt install git && sudo apt install libpng-dev
```

### Links e recursos adicionais

- [Guia super detalhado para fazer o VSCode funcionar com ESL no site de documentos da VSCode](https://code.visualstudio.com/docs/remote/wsl)

> - [Página da Microsoft Store para baixar o Ubuntu no Windows](https://www.microsoft.com/en-us/store/p/ubuntu/9nblggh4msv6)

Reviewer: ```suggestion
- [Página da Microsoft Store para baixar o Ubuntu no Windows](https://www.microsoft.com/pt-br/store/p/ubuntu/9nblggh4msv6)
```
